sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSQLInsertTestSuite.scala

```diff
@@ -21,5 +21,23 @@ import org.apache.spark.sql.SQLInsertTestSuite
 import org.apache.spark.sql.hive.test.TestHiveSingleton
 
 class HiveSQLInsertTestSuite extends SQLInsertTestSuite with TestHiveSingleton {
+
+  private var originalPartitionMode = ""
```

Comment:
I think you can just use `null` instead of an empty string.

```
private var originalPartitionMode: String = _
```

here and

```scala
spark.conf.get("hive.exec.dynamic.partition.mode", null)
```

later.

Reply:
You can also do something like:

```scala
Option(spark.conf.get("hive.exec.dynamic.partition.mode", null))
```

too. and later:

```scala
originalPartitionMode
  .map(v => spark.conf.set("hive.exec.dynamic.partition.mode", v)
  .getOrElse(spark.conf.unset("hive.exec.dynamic.partition.mode"))
```

up to you

Reply:
I forked this from HiveCharVarchar Test Suite. Shall we change there
 too

Reply:
No in this context because this is a follow-up of SPARK-32976. To touch other test suite, we need to make another JIRA.